unittests/rem_voting_test.cpp

```diff
@@ -656,4 +667,87 @@ BOOST_FIXTURE_TEST_CASE( rem_vote_weight_test, voting_tester ) {
 }
 
+BOOST_FIXTURE_TEST_CASE( resignation_test_case, voting_tester ) {
+    try {
+        // Create eosio.msig and eosio.token
+        create_accounts({N(eosio.msig), N(eosio.token), N(eosio.ram), N(eosio.ramfee), N(eosio.stake), N(eosio.vpay), N(eosio.bpay), N(eosio.saving) });
+
+        set_code_abi(N(eosio.msig),
+                     contracts::eosio_msig_wasm(),
+                     contracts::eosio_msig_abi().data());//, &eosio_active_pk);
+        set_code_abi(N(eosio.token),
+                     contracts::eosio_token_wasm(),
+                     contracts::eosio_token_abi().data()); //, &eosio_active_pk);
+
+        // Set privileged for eosio.msig and eosio.token
+        set_privileged(N(eosio.msig));
+        set_privileged(N(eosio.token));
+
+        // Verify eosio.msig and eosio.token is privileged
+        const auto& eosio_msig_acc = get<account_metadata_object, by_name>(N(eosio.msig));
+        BOOST_TEST(eosio_msig_acc.is_privileged() == true);
+        const auto& eosio_token_acc = get<account_metadata_object, by_name>(N(eosio.token));
+        BOOST_TEST(eosio_token_acc.is_privileged() == true);
+
+        // Create SYS tokens in eosio.token, set its manager as eosio
+        auto max_supply = core_from_string("1000000000.0000");
+        auto initial_supply = core_from_string("900000000.0000");
+        create_currency(N(eosio.token), config::system_account_name, max_supply);
+        // Issue the genesis supply of 1 billion SYS tokenbs to eosio.system
+        issue(N(eosio.token), config::system_account_name, config::system_account_name, initial_supply);
+
+        auto actual = get_balance(config::system_account_name);
+        BOOST_REQUIRE_EQUAL(initial_supply, actual);
+
+        // Create genesis accounts
+        for( const auto& a : rem_test_genesis ) {
+            create_account( a.name, config::system_account_name );
+        }
+        deploy_contract();
+
+        // Buy ram and stake cpu and net for each genesis accounts
+        for( const auto& acc : rem_test_genesis ) {
+            auto stake_quantity = acc.initial_balance - 1000;
+
+
+            auto r = delegate_bandwidth(N(eosio.stake), acc.name, asset(stake_quantity));
+            BOOST_REQUIRE( !r->except_ptr );
+        }
+
+        const auto whales_as_producers = { N(b1), N(whale1), N(whale3), N(whale2) };
+        for( const auto& producer : whales_as_producers ) {
+            register_producer(producer);
+        }
+
+        register_producer( N(proda));
+
+        // Vote for producers
+        auto votepro = [&]( account_name voter, vector<account_name> producers ) {
+            std::sort( producers.begin(), producers.end() );
+            base_tester::push_action(config::system_account_name, N(voteproducer), voter, mvo()
+                    ("voter",  name(voter))
+                    ("proxy", name(0) )
+                    ("producers", producers)
+            );
+        };
+
+        votepro( N(b1), { N(proda)} );
```

Comment:
@AndriiAndrusyshyn you can use the list from above to vote for `proda` 
```
for( const auto& producer : whales_as_producers ) {
    votepro( producer, { N(proda) };
}
```

p. s.
it's okay for `proda` to vote for himself